README.md

```diff
@@ -28,14 +28,245 @@ In addition, that format would include optional **signing** of the resources, wh
+followed by an HTTP response including at least the `Content-Type`,
+`Content-Location`, and `Date` headers, and one or
+more [signatures](#signatures).
+
+At least one of a manifest's signatures must be from a certificate that has a
+valid chain to a known root, and that is trusted to sign content from the
+[origin](https://html.spec.whatwg.org/multipage/browsers.html#concept-origin) of
+the manifest's `Content-Location`.
+
+If no signature gives the manifest authority to speak for its origin, or if any
+resource or sub-package doesn't match the description in the manifest, the whole
+package must fail to parse. TODO: Should there be a way to express that
+particular resources don't need to verify, in order to let the UA load the beginning of a package before the whole package has transferred?
+
+As a special case, if the package is being transferred from the manifest's
+origin under TLS, the UA may load it before checking that the resources match
+the manifest.
+
+TODO: Should we just extend
+the [Web App Manifest](https://www.w3.org/TR/appmanifest/), or does the fact
+that we're going to sign this rule out JSON?
+
+Resources are described by a line consisting of:
+
+```
+relative/url hashalgorithm-base64digest hashalgorithm2-base64digest2 ...
+```
+
+The URL is relative to the manifest's `Content-Location`.
+
+The hashes are in the format defined for the
+[`integrity` HTML attribute](https://w3c.github.io/webappsec-subresource-integrity/#the-integrity-attribute).
+This allows multiple hashes in order to provide agility in the face of future
+cryptographic discoveries. See also https://tools.ietf.org/html/rfc7696.
+
+The hashes are computed over the entire resource, including its length, HTTP
+headers, and body of the named resource. This differs
+from [SRI](https://w3c.github.io/webappsec-subresource-integrity), which only
+hashes the body. Note: This will usually prevent a package from relying on some
+of its contents being transferred as normal network responses, unless its author
+can guarantee the network won't change or reorder the headers.
+
+TODO: Do we need the length to be shown in the manifest, or is its presence at
+the beginning of the resource, included in the hash, enough?
+
+
+#### Sub-packages
+
+There are three possible forms of dependencies on sub-packages, of which we
+allow two. Note that a sub-package is or can be protected by its
+own [signature](#signatures), so if the main package trusts the sub-package's
+server, it could avoid specifying a version of the sub-package at all. However,
+this opens the main package up to downgrade attacks, where the sub-package is
+replaced by an older, vulnerable version, so we don't allow this option.
+
+If the main package wants to load either the sub-package it was built with or
+any upgrade, it can specify the date of the original sub-package:
+
+```
+https://package/manifest/url Date: Tue, 07 Feb 2017 01:05:54 GMT
```

Comment:
need to bikeshed the format here, HTTP Header on same line as other data seems an odd choice and hard to deal with.